substrate/bin/node/runtime/src/lib.rs

```diff
@@ -1460,7 +1460,7 @@ impl pallet_grandpa::Config for Runtime {
 
 parameter_types! {
 	pub const BasicDeposit: Balance = 10 * DOLLARS;       // 258 bytes on-chain
-	pub const FieldDeposit: Balance = 250 * CENTS;        // 66 bytes on-chain
+	pub const ByteDeposit: Balance = deposit(0, 1);
 	pub const SubAccountDeposit: Balance = 2 * DOLLARS;   // 53 bytes on-chain
```

Comment:
For Rococo and Westend we can leave them since we'll drop the pallet from the Relay Chains and configure it correctly in genesis of the parachain, but here (being the example runtime), I think we should use `deposit()` for the `BasicDeposit` and `SubAccountDeposit` configs using the smallest encoded size of a set identity.

Reply:
To answer the obvious questions first: the smallest identity you can set is an empty one, encoding to 9 bytes. The whole `Registration` `struct` that actually contains the identity would encode to a bit more, but it has stuff we don't charge for right now, such as the deposit held and the judgements.

That being said, I think of this deposit a bit differently. First off, even with the smallest identity, those 9 bytes will be accounted for because we take the encoded size of the whole `struct`, not just the data within it. Right now, the base deposit is way bigger than what it actually needs to be to store the data on chain, and I think that's an intentional barrier of entry into the pallet right now. I don't know if the deposits *need* to be this big because of some other economic reasons. My point is this: the storage cost should already be accounted for by the byte deposit, so the base deposit shouldn't depend on the `struct` size. It should instead be tied to some economic reason and if we can't find one, we can remove it or set it to `0` on our implementations and let downstream users decide if they want to add such a cost.

Reply:
I should have checked before commenting, but on Polkadot it's already configured [like I suggested](https://github.com/polkadot-fellows/runtimes/blob/7bd91e826e4b2d95f6d2ee7f6bb0f0f47d095859/relay/polkadot/src/lib.rs#L665). On Kusama, [not](https://github.com/polkadot-fellows/runtimes/blob/7bd91e826e4b2d95f6d2ee7f6bb0f0f47d095859/relay/kusama/src/lib.rs#L919).

By using `deposit`, it does actually come out to over 20 DOT:
```python
>>> def deposit(items, bytes):
...     return items * 20 * 10_000_000_000 + bytes * 100 * 100_000
... 
>>> deposit(1, 9) / 10**10
20.009
```

You aren't just paying for the size in bytes, you're paying for having a storage item at all, because more items increases the depth of the trie and therefore increases weights for all transactions since reads, writes, and trie hashing take longer.

For sure other users can configure this with a higher or lower `BasicDeposit` to subsidize or discourage its use, but I think the neutral config is to set it to `(1, min_size(Registration))`.

Reply:
The `BasicDeposit` for identity is now `deposit(1, min_size(Registration) - min_size(IdentityInfo)`, because the bytes in the identity info are already accounted for in the byte deposit. I didn't replicate the change for the subs deposit because if we were to do it, that would be another 2 storage items: one entry in `SuperOf` per sub identity of an account, one entry in `SubsOf` per account which uses the sub-account functionality. On top of that, we would have to also charge per byte and the price for a sub-identity would end up being about twice the price of setting one identity. Right now, the price of a sub-identity is 1/5 of an identity. I'd prefer to keep `SubAccountDeposit` as is, let me know what you think.